### PR TITLE
labelify the check_match errors

### PR DESCRIPTION
**Pull request by arielb1**
r? @jonathandturner

### CODE REVIEW

**Comment by jonathandturner** [comment]
@arielb1 - looks pretty good, though I notice you fix errors that other people have signed up for in the "help us fix errors" list: https://github.com/rust-lang/rust/issues/35233

We may want to see if they can be fixed by these new volunteers.

File: src/librustc_const_eval/check_match.rs [hunk 1]
```diff
@@ -235,12 +225,7 @@ fn check_expr(cx: &mut MatchCheckCtxt, ex: &hir::Expr) {
                 .flat_map(|arm| &arm.0)
                 .map(|pat| vec![wrap_pat(cx, &pat)])
                 .collect();
-            let match_span = Span {
-                lo: ex.span.lo,
-                hi: scrut.span.hi,
-                expn_id: ex.span.expn_id
-            };
-            check_exhaustive(cx, match_span, &matrix, source);
+            check_exhaustive(cx, ex.span, &matrix, source);
```
Comment (KiChjang) [inline src/librustc_const_eval/check_match.rs]: `ex.span` is what we had before, and what we need here is actually `scrut.span`.

Comment (arielb1) [inline src/librustc_const_eval/check_match.rs]: fixed

**Comment by arielb1** [comment]
closing this PR to see how this situation evolves.

**Pull request closed without merging**